### DISCUSSION TITLE
How to use Tearing plugin?

### DISCUSSION THREAD

**Discussion by RickYang1** [InfinyTech3D/SofaUnity#82] [Help - Using SofaUnity] [answered]
Hi,
    I added the TearingEngine component to the scn file, but when I imported the scn file in Unity, I encountered an error:
![image](https://github.com/InfinyTech3D/SofaUnity/assets/142986134/cd44d151-efee-40af-812b-8ee67af10de3)
This is my scn file：
`<?xml version="1.0" ?>

<Node name="root" dt="0.01" showBoundingTree="0" gravity="0 -10 0">
	<Node name="RequiredPlugins">
		<RequiredPlugin name="Sofa.Component.Collision.Detection.Algorithm"/> 
		<RequiredPlugin name="Sofa.Component.Collision.Detection.Intersection"/> 
		<RequiredPlugin name="Sofa.Component.Collision.Geometry"/> 
		<RequiredPlugin name="Sofa.Component.Collision.Response.Contact"/> 
		<RequiredPlugin name="Sofa.Component.Constraint.Projective"/> 
		<RequiredPlugin name="Sofa.Component.Engine.Select"/> 
		<RequiredPlugin name="Sofa.Component.IO.Mesh"/> 
		<RequiredPlugin name="Sofa.Component.LinearSolver.Iterative"/> 
		<RequiredPlugin name="Sofa.Component.Mapping.Linear"/> 
		<RequiredPlugin name="Sofa.Component.Mass"/> 
		<RequiredPlugin name="Sofa.Component.ODESolver.Backward"/> 
		<RequiredPlugin name="Sofa.Component.SceneUtility"/> 
		<RequiredPlugin name="Sofa.Component.SolidMechanics.FEM.Elastic"/> 
		<RequiredPlugin name="Sofa.Component.SolidMechanics.Spring"/> 
		<RequiredPlugin name="Sofa.Component.StateContainer"/> 
		<RequiredPlugin name="Sofa.Component.Topology.Container.Dynamic"/> 
		<RequiredPlugin name="Sofa.Component.Visual"/> 
		<RequiredPlugin name="Sofa.GL.Component.Rendering3D"/> 
		<RequiredPlugin name="Tearing"/>
        <RequiredPlugin name="Sofa.Component.MechanicalLoad"/> 
	</Node>
	
    <VisualStyle displayFlags="showVisual showBehaviorModels" />

    <DefaultAnimationLoop />
    <DefaultPipeline verbose="0" />
    <BruteForceBroadPhase/>
    <BVHNarrowPhase/>
    <CollisionResponse response="PenalityContactForceField" />
    <MinProximityIntersection name="Proximity" alarmDistance="0.8" contactDistance="0.5" />
    
    <Node name="SquareGravity_FEM">
        <EulerImplicitSolver name="cg_odesolver" printLog="false"  rayleighStiffness="0.1" rayleighMass="0.1" />
        <CGLinearSolver iterations="25" name="linear solver" tolerance="1.0e-9" threshold="1.0e-9" />
        <MeshGmshLoader name="meshLoader" filename="mesh/square3.msh" scale="10" createSubelements="true" />
        
        <TriangleSetTopologyContainer  name="Container" src="@meshLoader" fileTopology="" tags=" " />
        <TriangleSetTopologyModifier   name="Modifier" />
        <TriangleSetGeometryAlgorithms name="GeomAlgo"   template="Vec3d" />
        
        <MechanicalObject name="Mo"/>
        <DiagonalMass massDensity="2.0" />
        
        <BoxROI name="FixedSide" box="-0.1 9.9 0.1  10.1 10.1 -0.1" drawBoxes="1" />
        <FixedConstraint name="FixedConstraint" indices="@FixedSide.indices" />

        <TriangularFEMForceField name="FEM" youngModulus="300" poissonRatio="0.3" method="large" computePrincipalStress="1"/>
        <TriangularBendingSprings name="FEM-Bend" stiffness="300" />
        <PlaneForceField normal="0 1 0" d="-20" stiffness="100000" showPlane="1" showPlaneSize="20"/>
        
        <TriangleCollisionModel />
        
        <TearingEngine name="TearingEngine" input_position="@Mo.position" stressThreshold="10.0" 
            step="30" nbFractureMax="40" showFracturePath="1" fractureMaxLength="1.0"/>
			
        <Node >
            <OglModel name="Visual" color="red" />
            <IdentityMapping input="@.." output="@Visual" />
        </Node>
    </Node>
</Node>

Can I currently use the Tearing plugin in Unity? If not, what should I do to implement it?
Thank you in advance.

**Comment by epernod** [comment]
Hi,

the tearing will be part of the next release, here is the PR related to the integration: https://github.com/InfinyTech3D/SofaUnity/pull/94
If you want to test before, you need to add the Tearing plugin to the list of loaded plugin in Unity. [This change in the PR](https://github.com/InfinyTech3D/SofaUnity/pull/94/files#diff-7df2a1a09d6dfa0b2d65830f76cda348cd20b73b5b9f43403757cf7b9e185de2)

Best regards

**Reply by RickYang1** [reply]
Hi，
    Thank you very much for your answer. Now I have resolved my error according to the method you provided, but when I run in Unity, the tearing did not occur：

https://github.com/InfinyTech3D/SofaUnity/assets/142986134/1d3158e4-5338-429d-89ab-e52e7ad8d35b

But it's possible in runsofa：

https://github.com/InfinyTech3D/SofaUnity/assets/142986134/c420cc89-fa5b-4731-92d3-9426d9c110fd

    Do I need to make any further modifications or obtain an available license.
    Thanks in advance.

**Reply by epernod** [reply, marked as answer]
which version do you use? Are you compiling the plugin on your side?
you might need this fix: https://github.com/InfinyTech3D/Tearing/commit/72be5ecf2998547bf793d5913fc6db69a7ac9400

**Reply by RickYang1** [reply]
> which version do you use? Are you compiling the plugin on your side? you might need this fix: https://github.com/InfinyTech3D/Tearing/commit/72be5ecf2998547bf793d5913fc6db69a7ac9400

Thank you for your help. Currently, the testing is good. Looking forward to the next release. 
Cheers
